modules/fedimint-mint-client/src/output.rs

```diff
@@ -220,68 +212,66 @@ impl MintOutputStatesCreated {
+        // we combine the shares, finalize the issuance request with the blind signature
+        // and store the resulting note in the database
+
+        let created = match old_state.state {
+            MintOutputStates::Created(created) => created,
             _ => panic!("Unexpected prior state"),
         };
 
-        // if the query obtained a threshold of valid blind signature shares, we combine
-        // the shares, finalize the issuance request with the blind signature
-        // and store the resulting note in the database
-        let note_res = output_outcomes_result.and_then(|blind_signature_shares| {
-            match mint_keys.tier(&amount) {
-                Ok(amount_key) => issuance_request
-                    .finalize(
-                        aggregate_signature_shares(
-                            &blind_signature_shares
-                                .iter()
-                                .map(|(peer, share)| {
-                                    let share = share.ensure_v0_ref().expect(
-                                    "We only process output outcome versions created by ourselves",
-                                );
-                                    (peer.to_usize() as u64 + 1, share.0)
-                                })
-                                .collect(),
-                        ),
-                        *amount_key,
-                    )
-                    .map_err(|e| e.to_string()),
-                Err(error) => Err(NoteFinalizationError::InvalidAmountTier(error.0).to_string()),
-            }
-        });
-
-        match note_res {
-            Ok(note) => {
-                if let Some(note) = dbtx
-                    .module_tx()
-                    .insert_entry(
-                        &NoteKey {
-                            amount,
-                            nonce: note.nonce(),
-                        },
-                        &note,
-                    )
-                    .await
-                {
-                    error!(
-                        ?note,
-                        "E-cash note was replaced in DB, this should never happen!"
-                    )
-                }
-
-                MintOutputStateMachine {
-                    common: old_state.common,
-                    state: MintOutputStates::Succeeded(MintOutputStatesSucceeded { amount }),
-                }
-            }
-            Err(error) => MintOutputStateMachine {
+        let agg_blind_signature = aggregate_signature_shares(
+            &blinded_signature_shares
+                .into_iter()
+                .map(|(peer, share)| (peer.to_usize() as u64 + 1, share))
+                .collect(),
+        );
+
+        let amount_key = tbs_pks
+            .tier(&created.amount)
+            .expect("We obtained this amount from tbs_pks when we created the output");
+
+        // this implies that the mint client config's public keys are inconsistent
```

Comment:
So maybe in the future we can just check config consistency when joining the federation and then expect here?

Reply:
Yes we could